### PR TITLE
[NLL] Fix ICE when a borrow wrapped in a temporary is used after dropped

### LINKED ISSUE #47646: ICE on nightly with -Znll-dump-cause and match on a tuple
On nightly `rustc` 2018-01-20 the following code snippet

```rust
#![feature(nll)]

use std::collections::BinaryHeap;

fn main() {
    let mut heap: BinaryHeap<i32> = BinaryHeap::new();
    let borrow = heap.peek_mut();

    match (borrow, ()) {
        (Some(_), ()) => {
            println!("{:?}", heap);
        },

        _ => {	}
    };
}
```
causes an ICE when run as `rustc +nightly -Znll-dump-cause src/main.rs` (panic in `explain_why_borrow_contains_point`)

<details>
<summary>Backtrace</summary>

```
   9: core::panicking::panic
             at libcore/panicking.rs:51
  10: rustc_mir::borrow_check::nll::explain_borrow::<impl rustc_mir::borrow_check::MirBorrowckCtxt<'cx, 'gcx, 'tcx>>::explain_why_borrow_contains_point
  11: rustc_mir::borrow_check::error_reporting::<impl rustc_mir::borrow_check::MirBorrowckCtxt<'cx, 'gcx, 'tcx>>::report_conflicting_borrow
  12: rustc_mir::borrow_check::MirBorrowckCtxt::access_place
  13: <rustc_mir::borrow_check::MirBorrowckCtxt<'cx, 'gcx, 'tcx> as rustc_mir::dataflow::DataflowResultsConsumer<'cx, 'tcx>>::visit_statement_entry
  14: rustc::ty::context::tls::enter
  15: rustc::infer::InferCtxtBuilder::enter
  16: rustc_mir::borrow_check::mir_borrowck
  17: rustc::ty::maps::<impl rustc::ty::maps::queries::mir_borrowck<'tcx>>::compute_result
  18: rustc::dep_graph::graph::DepGraph::with_task_impl
  19: rustc_errors::Handler::track_diagnostics
  20: rustc::ty::maps::plumbing::<impl rustc::ty::context::TyCtxt<'a, 'gcx, 'tcx>>::cycle_check
  21: rustc::ty::maps::<impl rustc::ty::maps::queries::mir_borrowck<'tcx>>::force
  22: rustc::ty::maps::<impl rustc::ty::maps::queries::mir_borrowck<'tcx>>::try_get
  23: rustc::ty::maps::TyCtxtAt::mir_borrowck
  24: rustc::ty::maps::<impl rustc::ty::context::TyCtxt<'a, 'tcx, 'lcx>>::mir_borrowck
  25: rustc_driver::driver::phase_3_run_analysis_passes::{{closure}}::{{closure}}
  26: <std::thread::local::LocalKey<T>>::with
  27: <std::thread::local::LocalKey<T>>::with
  28: rustc::ty::context::TyCtxt::create_and_enter
  29: rustc_driver::driver::compile_input
  30: rustc_driver::run_compiler
```
</details>

<details>
<summary>`rustc` version details</summary>

```
rustc 1.25.0-nightly (15a1e2844 2018-01-20)
binary: rustc
commit-hash: 15a1e2844dfea7850be5c6c901b67ceff370b0eb
commit-date: 2018-01-20
host: x86_64-unknown-linux-gnu
release: 1.25.0-nightly
LLVM version: 4.0
```
</details>

### PR DESCRIPTION
**Pull request by spastorino**
Fixes #47646

r? @nikomatsakis

### CODE REVIEW

**Comment by TimNN** [comment]
Your PR [failed on Travis](https://travis-ci.org/rust-lang/rust/jobs/364218720) ([raw log](https://api.travis-ci.org/v3/job/364218720/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
$ git fetch origin +refs/pull/49808/merge:
error: RPC failed; curl 18 transfer closed with outstanding read data remaining
fatal: The remote end hung up unexpectedly
The command "eval git fetch origin +refs/pull/49808/merge: " failed. Retrying, 2 of 3.
---
[00:00:45] configure: rust.quiet-tests     := True
---
[00:05:06] tidy error: /checkout/src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs:76: line longer than 100 chars
[00:05:07] some tidy checks failed
[00:05:07]
[00:05:07]
[00:05:07] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/tidy" "/checkout/src" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "--no-vendor" "--quiet"
[00:05:07] expected success, got: exit code: 1
[00:05:07]
[00:05:07]
[00:05:07] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test src/tools/tidy
[00:05:07] Build completed unsuccessfully in 0:01:54
[00:05:07] make: *** [tidy] Error 1
[00:05:07] Makefile:79: recipe for target 'tidy' failed
---
$ ls -lat $HOME/Library/Logs/DiagnosticReports/
ls: cannot access /home/travis/Library/Logs/DiagnosticReports/: No such file or directory
travis_time:end:02e8f10a:start=1523294250381205128,finish=1523294250387949449,duration=6744321
travis_fold:end:after_failure.2
travis_fold:start:after_failure.3
travis_time:start:03cd043a
$ find $HOME/Library/Logs/DiagnosticReports -type f -name '*.crash' -not -name '*.stage2-*.crash' -not -name 'com
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

**Comment by TimNN** [comment]
Your PR [failed on Travis](https://travis-ci.org/rust-lang/rust/jobs/364228999) ([raw log](https://api.travis-ci.org/v3/job/364228999/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
[00:01:05] configure: rust.quiet-tests     := True
---
[00:16:56] error: unused variable: `temporary`
[00:16:56]   --> librustc_mir/borrow_check/nll/explain_borrow/mod.rs:73:41
[00:16:56]    |
[00:16:56] 73 |                                     let temporary = mir.local_decls[local];
[00:16:56]    |                                         ^^^^^^^^^ help: consider using `_temporary` instead
---
[00:16:56]    = note: #[deny(unused_variables)] implied by #[deny(warnings)]
---
[00:16:56]   process didn't exit successfully: `/checkout/obj/build/bootstrap/debug/rustc --crate-name rustc_mir librustc_mir/lib.rs --color always --error-format json --crate-type dylib --emit=dep-info,link -C prefer-dynamic -C opt-level=2 -C metadata=97e644ee6c469e67 -C extra-filename=-97e644ee6c469e67 --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/release/deps --extern syntax_pos=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/libsyntax_pos-9f3518d56a01456f.so --extern rustc=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc-23f5dcecdda8feb4.so --extern serialize=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/libserialize-c04ded78717d5d67.so --extern serialize=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/libserialize-c04ded78717d5d67.rlib --extern rustc_data_structures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_data_structures-93cb1ddd29ab61a4.so --extern arena=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/libarena-f2778bd6cd1c5bdd.so --extern graphviz=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/libgraphviz-0bde40de32995f14.so --extern bitflags=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/libbitflags-9866e194db82a141.rlib --extern rustc_back=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_back-861dba9fe03aa669.so --extern log=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/liblog-3e35efb9e5bc7a9a.rlib --extern rustc_apfloat=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_apfloat-1299638d641ea770.rlib --extern syntax=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/libsyntax-d3b6fcf798f7d22a.so --extern rustc_const_math=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_const_math-895f7ddc4467bb8d.so --extern log_settings=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/liblog_settings-70f7ed359bd54256.rlib --extern byteorder=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/libbyteorder-ca61dfec7c40c4d4.rlib --extern rustc_errors=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_errors-609e2421d03f9c9a.so -L native=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/build/backtrace-sys-8b35e3c2ea935fab/out/.libs -L native=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/build/miniz-sys-63734d0048644b22/out` (exit code: 101)
[00:16:56] warning: build failed, waiting for other jobs to finish...
[00:19:02] error: build failed
[00:19:02] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "build" "--target" "x86_64-unknown-linux-gnu" "--release" "--locked" "--color" "always" "--features" " jemalloc" "--manifest-path" "/checkout/src/rustc/Cargo.toml" "--message-format" "json"
[00:19:02] expected success, got: exit code: 101
[00:19:02] thread 'main' panicked at 'cargo must succeed', bootstrap/compile.rs:1085:9
[00:19:02] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[00:19:02] travis_fold:end:stage0-rustc
[00:19:02] travis_time:end:stage0-rustc:start=1523295798894120507,finish=1523296583933321570,duration=785039201063
[00:19:02] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap build
[00:19:02] Build completed unsuccessfully in 0:13:19
[00:19:02] Makefile:28: recipe for target 'all' failed
[00:19:02] make: *** [all] Error 1
---
121664 ./obj/build/bootstrap/debug/incremental/bootstrap-351vorei3hhuv/s-ezyobz2ivp-1l4zdz-179sb4yv5t1yk
---
65620 ./obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/depot access /home/travis/Library/Logs/DiagnosticReports/: No such file or directory
travis_time:end:03539adc:start=1523296584740592199,finish=1523296584755963244,duration=15371045
travis_fold:end:after_failure.2
travis_fold:start:after_failure.3
travis_time:start:005badb5
$ find $HOME/Library/Logs/DiagnosticReports -type f -name '*.crash' -not -name '*.stage2-*.crash' -not -name 'com.apple.CoreSimulator.CoreSimulatorService-*.crash' -exec printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" {} \; -exec head -750 {} \; -exec echo travis_fold":"end:crashlog \; || true
find: `/home/travis/Library/Logs/DiagnosticReports': No such file or directory
travis_time:end:005badb5:start=1523296584768768168,finish=1523296584783318627,duration=14550459
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:20809be2
$ dmesg | grep -i kill
[   13.817240] init: failsafe main process (1093) killed by TERM signal
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

File: src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs [hunk 1]
```diff
@@ -59,17 +60,33 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
...
+
+                                    if locations.len() == 1 {
+                                        err.span_label(
+                                            mir.source_info(locations[0]).span,
+                                            "borrow may end up in a temporary, created here"
+                                        );
+                                    }
+
+                                    err.span_label(
+                                        mir.source_info(p).span,
```
Comment (nikomatsakis) [inline src/librustc_mir/borrow_check/nll/explain_borrow/mod.rs]: actually maybe we can just do `mir.local_decls[local].source_info.span` here and not bother to search for locals?

**Comment by TimNN** [comment]
Your PR [failed on Travis](https://travis-ci.org/rust-lang/rust/jobs/364278988) ([raw log](https://api.travis-ci.org/v3/job/364278988/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
[00:01:08] configure: rust.quiet-tests     := True
---
[00:42:00] ............................................................................i.......................
[00:42:05] ...................i................................................................................
[00:42:09] ....................................................................................................
[00:42:13] ....................................................................................................
[00:42:17] ....................................................................................................
[00:42:21] ....................................................................................................
[00:42:27] ....................................................................................................
[00:42:33] ...............................................................F....................................
[00:42:39] ..................................................................................................i.
[00:42:46] ........................................................................i...........................
---
[00:43:07] 1 error[E0502]: cannot borrow `heap` as immutable because it is also borrowed as mutable
[00:43:07] -   --> $DIR/issue-47646.rs:23:30
[00:43:07] +   --> $DIR/issue-47646.rs:22:30
[00:43:07] 3    |
[00:43:07] 4 LL |     let borrow = heap.peek_mut();
[00:43:07] 5    |                  ---- mutable borrow occurs here
[00:43:07]
[00:43:07]
[00:43:07] The actual stderr differed from the expected stderr.
[00:43:07] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issue-47646.stderr
[00:43:07] To update references, run this command from build directory:
[00:43:07] /checkout/src/test/ui/update-references.sh '/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui' 'issue-47646.rs'
[00:43:07]
[00:43:07] error: 1 errors occurred comparing output.
[00:43:07] status: exit code: 101
[00:43:07] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/issue-47646.rs" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/issue-47646.stage2-x86_64-unknown-linux-gnu" "-Crpath" "-O" "-Zmiri" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknowne","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issue-47646.rs","byte_start":624,"byte_end":628,"line_start":18,"line_end":18,"column_start":18,"column_end":22,"is_primary":false,"text":[{"text":"    let borrow = heap.peek_mut();","highlight_start":18,"highlight_end":22}],"label":"mutable borrow occurs here","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issue-47646.rs","byte_start":652,"byte_end":664,"line_start":20,"line_end":20,"column_start":11,"column_end":23,"is_primary":false,"text":[{"text":"    match (borrow, ()) {","highlight_start":11,"highlight_end":23}],"label":"borrow may end up in a temporary, created here","suggested_replacement":null,"expansion":null},{"file_name":"/checkout/src/test/ui/issue-47646.rs","byte_start":805,"byte_end":806,"line_start":25,"line_end":25,"column_start":6,"column_end":7,"is_primary":false,"text":[{"text":"    };","highlight_start":6,"highlight_end":7}],"label":"temporary later dropped here, potentially using the reference","suggested_replacement":null,"expansion":null}],"children":[],"rendered":"error[E0502]: cannot borrow `heap` as immutable because it is also borrowed as mutable\n  --> /checkout/src/test/ui/issue-47646.rs:22:30\n   |\nLL |     let borrow = heap.peek_mut();\n   |                  ---- mutable borrow occurs here\nLL | \nLL |     match (borrow, ()) {\n   |           ------------ borrow may end up in a temporary, created here\nLL |         (Some(_), ()) => {\nLL |             println!(\"{:?}\", heap); //~ ERROR cannot borrow `heap` as immutable\n   |                              ^^^^ immutable borrow occurs here\n...\nLL |     };\n   |      - temporary later dropped here, potentially using the reference\n\n"}
[00:43:07] {"message":"aborting due to previous error","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to previous error\n\n"}
[00:43:07] {"message":"For more information about this error, try `rustc --explain E0502`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0502`.\n"}
---
[00:43:07] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-3.9/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zmiri -Zunstable-options " "--target-rustcflags" "-Crpath -O -Zmiri -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "3.9.1\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
[00:43:07] expected success, got: exit code: 101
[00:43:07]
[00:43:07]
[00:43:07] failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
[00:43:07] Build completed unsuccessfully in 0:02:14
[00:43:07] make: *** [check] Error 1
[00:43:07] Makefile:58: recipe for target 'check' failed
---
121664 ./obj/build/bootstrap/debug/incremental/bootstrap-351vorei3hhuv/s-ezyrm00ew0-1w34897-179sb4yv5t1yk
110164 ./obj/build/x86_64-unknown-linux-gnu/stage0-tools/x86_64-unknown-linux-gnu
110160 ./obj/build/x86_64-unknown-linux-gnu/stage0-tools/x86_64-unknown-linux-gnu/release
106248 ./obj/build/x86_64-unknown-linux-gnu/stage0-tools/x86_64-unknown-linux-gnu/release/deps
102808 ./obj/build/x86_64-unknown-linux-gnu/stage0/lib/rustlib/x86_64-unknown-linux-gnu/codegen-backends
102732 ./obj/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/incremental/core-31lccp6wy7orz
102728 ./obj/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/incremental/core-31lccp6wy7orz/s-ezysm8owr7-1d235fp-6xnnylbd6ghy
100676 ./obj/build/bootstrap/debug/incremental/bootstrap-zki88qpb4pjm
100672 ./obj/build/bootstrap/debug/incremental/bootstrap-zki88qpb4pjm/s-ezysofdf10-4y7uev-1raxz7drp4c1e
---
$ find $HOME/Library/Logs/DiagnosticReports -type f -name '*.crash' -not -name '*.stage2-*.crash' -not -name 'com.apple.CoreSimulator.CoreSimulatorService-*.crash' -exec printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" {} \; -exec head -750 {} \; -exec echo travis_fold":"end:crashlog \; || true
find: `/home/travis/Library/Logs/DiagnosticReports': No such file or directory
travis_time:end:1d7c31f8:start=1523305171498746801,finish=1523305171505290894,duration=6544093
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:09f52b4e
$ dmesg | grep -i kill
[   12.310630] init: failsafe main process (1092) killed by TERM signal
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

**Comment by TimNN** [comment]
Your PR [failed on Travis](https://travis-ci.org/rust-lang/rust/jobs/364566771) ([raw log](https://api.travis-ci.org/v3/job/364566771/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
[00:00:01] tar: Skipping to next header
[00:00:02] src/ci/init_repo.sh: line 46:  3745 Segmentation fault      (core dumped) tar -C $module --strip-components=1 -xf $cached
---
[00:00:24] error during connect: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.32/images/load?quiet=1: EOF
[00:00:24] Downloaded containers:\n
[00:00:24] Attempting with retry: docker build --rm -t rust-ci -f /home/travis/build/rust-lang/rust/src/ci/docker/dist-powerpc64-linux/Dockerfile /home/travis/build/rust-lang/rust/src/ci/docker
[00:00:24] Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
[00:00:24] time="2018-04-10T11:47:28Z" level=error msg="failed to dial gRPC: cannot connect to the Docker daemon. Is 'docker daemon' running on this host?: dial unix /var/run/docker.sock: connect: connection refused"
[00:00:24] time="2018-04-10T11:47:28Z" level=error msg="Can't add file /home/travis/build/rust-lang/rust/src/ci/docker/armhf-gnu/vexpress_config to tar: io: read/write on closed pipe"
[00:00:24] time="2018-04-10T11:47:28Z" level=error msg="Can't close tar writer: io: read/write on closed pipe"
[00:00:24] Command failed. Attempt 2/5:
[00:00:25] time="2018-04-10T11:47:29Z" level=error msg="failed to dial gRPC: unable to upgrade to h2c, received 501"
[00:00:25] time="2018-04-10T11:47:29Z" level=error msg="Can't add file /home/travis/build/rust-lang/rust/src/ci/docker/dist-armhf-linux/arm-linux-gnueabihf.config to tar: io: read/write on closed pipe"
[00:00:25] time="2018-04-10T11:47:29Z" level=error msg="Can't close tar writer: io: read/write on closed pipe"
[00:00:25] context canceled
---
[00:01:05] Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
[00:01:05] time="2018-04-10T11:48:10Z" level=error msg="failed to dial gRPC: cannot connect to the Docker daemon. Is 'docker daemon' running on this host?: dial unix /var/run/docker.sock: connect: connection refused"
[00:01:05] Command failed. Attempt 5/5:
[00:01:05] Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
[00:01:05] The command has failed after 5 attempts.
---
$ ls -lat $HOME/Library/Logs/DiagnosticReports/
ls: cannot access /home/travis/Library/Logs/DiagnosticReports/: No such file or directory
travis_time:end:23008758:start=1523360890406521546,finish=1523360890414622333,duration=8100787
travis_fold:end:after_failure.2
travis_fold:start:after_failure.3
travis_time:start:069eaaaf
$ find $HOME/Library/Logs/DiagnosticReports -type f -name '*.crash' -not -name '*.stage2-*.crash' -not -name 'com.apple.CoreSimulator.CoreSimulatorService-*.crash' -exec printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" {} \; -exec head -750 {} \; -exec echo travis_fold":"end:crashlog \; || true
find: `/home/travis/Library/Logs/DiagnosticReports': No such file or directory
travis_time:end:069eaaaf:start=1523360890420113670,finish=1523360890425995570,duration=5881900
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:13913500
$ dmesg | grep -i kill
[   10.573652] init: failsafe main process (1095) killed by TERM signal
[   33.307545] init: docker main process (1158) killed by ILL signal
[  133.074003] init: docker main process (4350) killed by SEGV signal
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

**Comment by TimNN** [comment]
Your PR [failed on Travis](https://travis-ci.org/rust-lang/rust/jobs/364566771) ([raw log](https://api.travis-ci.org/v3/job/364566771/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
[00:00:01] tar: Skipping to next header
[00:00:02] src/ci/init_repo.sh: line 46:  3745 Segmentation fault      (core dumped) tar -C $module --strip-components=1 -xf $cached
---
[00:00:24] error during connect: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.32/images/load?quiet=1: EOF
[00:00:24] Downloaded containers:\n
[00:00:24] Attempting with retry: docker build --rm -t rust-ci -f /home/travis/build/rust-lang/rust/src/ci/docker/dist-powerpc64-linux/Dockerfile /home/travis/build/rust-lang/rust/src/ci/docker
[00:00:24] Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
[00:00:24] time="2018-04-10T11:47:28Z" level=error msg="failed to dial gRPC: cannot connect to the Docker daemon. Is 'docker daemon' running on this host?: dial unix /var/run/docker.sock: connect: connection refused"
[00:00:24] time="2018-04-10T11:47:28Z" level=error msg="Can't add file /home/travis/build/rust-lang/rust/src/ci/docker/armhf-gnu/vexpress_config to tar: io: read/write on closed pipe"
[00:00:24] time="2018-04-10T11:47:28Z" level=error msg="Can't close tar writer: io: read/write on closed pipe"
[00:00:24] Command failed. Attempt 2/5:
[00:00:25] time="2018-04-10T11:47:29Z" level=error msg="failed to dial gRPC: unable to upgrade to h2c, received 501"
[00:00:25] time="2018-04-10T11:47:29Z" level=error msg="Can't add file /home/travis/build/rust-lang/rust/src/ci/docker/dist-armhf-linux/arm-linux-gnueabihf.config to tar: io: read/write on closed pipe"
[00:00:25] time="2018-04-10T11:47:29Z" level=error msg="Can't close tar writer: io: read/write on closed pipe"
[00:00:25] context canceled
---
[00:01:05] Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
[00:01:05] time="2018-04-10T11:48:10Z" level=error msg="failed to dial gRPC: cannot connect to the Docker daemon. Is 'docker daemon' running on this host?: dial unix /var/run/docker.sock: connect: connection refused"
[00:01:05] Command failed. Attempt 5/5:
[00:01:05] Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
[00:01:05] The command has failed after 5 attempts.
---
$ ls -lat $HOME/Library/Logs/DiagnosticReports/
ls: cannot access /home/travis/Library/Logs/DiagnosticReports/: No such file or directory
travis_time:end:23008758:start=1523360890406521546,finish=1523360890414622333,duration=8100787
travis_fold:end:after_failure.2
travis_fold:start:after_failure.3
travis_time:start:069eaaaf
$ find $HOME/Library/Logs/DiagnosticReports -type f -name '*.crash' -not -name '*.stage2-*.crash' -not -name 'com.apple.CoreSimulator.CoreSimulatorService-*.crash' -exec printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" {} \; -exec head -750 {} \; -exec echo travis_fold":"end:crashlog \; || true
find: `/home/travis/Library/Logs/DiagnosticReports': No such file or directory
travis_time:end:069eaaaf:start=1523360890420113670,finish=1523360890425995570,duration=5881900
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:13913500
$ dmesg | grep -i kill
[   10.573652] init: failsafe main process (1095) killed by TERM signal
[   33.307545] init: docker main process (1158) killed by ILL signal
[  133.074003] init: docker main process (4350) killed by SEGV signal
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

**Pull request merged**